sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -176,13 +91,16 @@ def _insert_overwrite_by_condition(
         source_queries: t.List[SourceQuery],
         columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
         where: t.Optional[exp.Condition] = None,
+        insert_overwrite_strategy_override: t.Optional[InsertOverwriteStrategy] = None,
     ) -> None:
         """
         Some engines do not support `INSERT OVERWRITE` but instead support
         doing an "INSERT OVERWRITE" using a Merge expression but with the
         predicate being `False`.
         """
         columns_to_types = columns_to_types or self.columns(table_name)
+        if not where:
+            where = exp.true()
```

Comment:
why is this needed? where=None should just mean no where statement right

Reply:
I think some engines require a condition on an overwrite? I will test.

Reply:
ah, i see, but it wasn't there before which is why i was wondering why it was added

Reply:
Tests ran fine without it so I will go ahead and remove.